File: Python/practical/6b.py
# Program to find years, weeks, days

def yearsWkDays(days):
    time = []
    time.append(days // 365)
    time.append((days - time[0] * 365) // 7)
    time.append(days - time[0] * 365 - time[1] * 7)
    
    return time

days = int(input("Enter the raw days: "))
print(f"{days} days = {yearsWkDays(days)[0]}:{yearsWkDays(days)[1]}:{yearsWkDays(days)[2]}")

File: C/SEM I/Practical/Done/6b.c
/* Write a C program to convert a given integer (in days) to years, weeks and days */

#include <stdio.h>
#include <conio.h>
#include <stdlib.h>

int count = 0;

int main(void) {
	int rep = 1;

	while (rep == 1) {
		int raw, y, w, d;

		printf("Enter the total days: ");
		scanf("%d", & raw);

		y = raw / 365;
		w = (raw - (365 * y)) / 7;
		d = (raw - (365 * y) - (w * 7));

		printf("%d days is %d years, %d weeks and %d days", raw, y, w, d);

		printf("\nRepeat Program?: ");
		scanf("%d", & rep);

		count++;

		if (count >= 50) {
			printf("Abnormal END\n");

			getch();
			exit(0);
		}

	}

	printf("Normal END\n");

	getch();
	return 0;
}